save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/topbar/TopBarLinks.kt

```diff
@@ -0,0 +1,127 @@
+import com.saveourtool.save.validation.FrontendRoutes
+
+import csstype.ClassName
+import csstype.Width
+import csstype.rem
+import history.Location
+import js.core.jso
+import react.FC
+import react.Props
+import react.dom.aria.AriaRole
+import react.dom.aria.ariaExpanded
+import react.dom.aria.ariaLabelledBy
+import react.dom.html.ReactHTML.a
+import react.dom.html.ReactHTML.div
+import react.dom.html.ReactHTML.li
+import react.dom.html.ReactHTML.ul
+import react.router.dom.Link
+import react.router.useNavigate
+import react.useState
+
+import kotlinx.browser.window
+
+val topBarLinks = topBarLinks()
+
+/**
+ * [Props] of the top bar links component
+ */
+external interface TopBarLinksProps : Props {
+    /**
+     * Is location
+     */
+    var location: Location
+}
+
+/**
+ * @property hrefAnchor is link
+ * @property width is width of the link text
+ * @property text is link text
+ */
+data class TopBarLink(
+    val hrefAnchor: String,
+    val width: Width,
+    val text: String,
+)
+
+/**
+ * Displays the static links that do not depend on the url
+ */
+@Suppress("MAGIC_NUMBER", "LongMethod", "TOO_LONG_FUNCTION")
+private fun topBarLinks() = FC<TopBarLinksProps> { props ->
+    val navigate = useNavigate()
+    val (isDemoDropdownActive, setIsDemoDropdownActive) = useState(false)
+
+    ul {
+        className = ClassName("navbar-nav mx-auto")
+        li {
+            className = ClassName("nav-item dropdown no-arrow")
+            style = jso {
+                width = 5.rem
+            }
+            a {
+                className = ClassName("nav-link dropdown-toggle text-light")
+                asDynamic()["data-toggle"] = "dropdown"
+                ariaExpanded = false
+                id = "demoDropdown"
+                role = "button".unsafeCast<AriaRole>()
+                +"Demo"
+                onClickCapture = { _ ->
+                    setIsDemoDropdownActive { !it }
+                }
+            }
+            div {
+                className = ClassName("dropdown-menu dropdown-menu-right shadow animated--grow-in${if (isDemoDropdownActive) " show" else "" }")
+                ariaLabelledBy = "demoDropdown"
+                val diktatDemoHref = "/${FrontendRoutes.DEMO.path}/diktat"
+                dropdownEntry(null, "Diktat", window.location.href.contains(diktatDemoHref)) { attrs ->
+                    attrs.onClick = {
+                        setIsDemoDropdownActive(false)
+                        navigate(to = diktatDemoHref)
+                    }
+                }
+                val cpgDemoHref = "/${FrontendRoutes.DEMO.path}/cpg"
+                dropdownEntry(null, "CPG", window.location.href.contains(cpgDemoHref)) { attrs ->
+                    attrs.onClick = {
+                        setIsDemoDropdownActive(false)
+                        navigate(to = cpgDemoHref)
+                    }
+                }
+            }
+        }
+        listOf(
```

Comment:
If you don't need a **container** and only need to **iterate** over multiple items, use `sequenceOf()`.

Reply:
Done [faec44c](https://github.com/saveourtool/save-cloud/pull/1540/commits/faec44ca7ee13b63b5e76ba2be295039cdb7c3e1)